benchmark/scripts/test_Benchmark_Driver.py

```diff
@@ -279,246 +262,219 @@ def setUp(self):
+             '--meta', '--quantile=20', '--delta')), 3)
         self.assertEqual(r.num_samples, 3)  # results are merged
 
     def test_run_and_log(self):
         def mock_run(test):
-            self.assertEqual(test, "b1")
+            self.assertEqual(test, 'b1')
             return PerformanceTestResult(
-                "3,b1,5,101,1,1,1,1,888".split(","),
-                quantiles=True,
-                delta=True,
-                memory=True,
-            )
-
-        driver = BenchmarkDriver(tests=["b1"], args=Stub(output_dir=None))
+                '3,b1,5,101,1,1,1,1,888'.split(','),
+                quantiles=True, delta=True, memory=True)
+        driver = BenchmarkDriver(tests=['b1'], args=Stub(output_dir=None))
         driver.run_independent_samples = mock_run  # patching
 
         with captured_output() as (out, _):
             log = driver.run_and_log()
 
-        header = (
-            "#,TEST,SAMPLES,MIN(μs),Q1(μs),MEDIAN(μs),Q3(μs),MAX(μs)," + "MAX_RSS(B)\n"
-        )
-        csv_log = "3,b1,5,101,102,103,104,105,888\n"
+        header = '#,TEST,SAMPLES,MIN(μs),Q1(μs),MEDIAN(μs),Q3(μs),MAX(μs),' +\
+            'MAX_RSS(B)\n'
+        csv_log = '3,b1,5,101,102,103,104,105,888\n'
         self.assertEqual(log, None)
         self.assertEqual(
             out.getvalue(),
-            header + csv_log + "\n" + "Total performance tests executed: 1\n",
-        )
+            header +
+            csv_log +
+            '\n' +
+            'Total performance tests executed: 1\n')
 
         with captured_output() as (out, _):
             log = driver.run_and_log(csv_console=False)
 
         self.assertEqual(log, header + csv_log)
         self.assertEqual(
             out.getvalue(),
-            "  # TEST                                     SAMPLES MIN(μs)"
-            + " Q1(μs) MEDIAN(μs) Q3(μs) MAX(μs) MAX_RSS(B)\n"
-            + "  3 b1                                             5     101"
-            + "    102        103    104     105        888\n"
-            + "\n"
-            + "Total performance tests executed: 1\n",
-        )
+            '  # TEST                                     SAMPLES MIN(μs)' +
+            ' Q1(μs) MEDIAN(μs) Q3(μs) MAX(μs) MAX_RSS(B)\n' +
+            '  3 b1                                             5     101' +
+            '    102        103    104     105        888\n' +
+            '\n' +
+            'Total performance tests executed: 1\n')
 
     def test_log_results(self):
         """Create log directory if it doesn't exist and write the log file."""
-
         def assert_log_written(out, log_file, content):
-            self.assertEqual(out.getvalue(), "Logging results to: " + log_file + "\n")
-            with open(log_file, "rU") as f:
+            self.assertEqual(out.getvalue(),
+                             'Logging results to: ' + log_file + '\n')
+            with open(log_file, 'r') as f:
```

Comment:
The `'rU'` mode's behavior is the default in Python 3, and explicitly setting this mode is deprecated (a warning is printed to stderr).

However, I'm not sure this change won't affect Python 2 compatibility. In #30085 I've defensively added a version check:

```python
if sys.version_info < (3, 0):
    openmode = "rU"
else:
    openmode = "r"
with open(log_file, openmode) as f:
    # ...
```

Are you sure this is okay to just use `'r'` here?

Reply:
All test passed with python3 in shebang, so I think the answer is yes.

Reply:
Have you tested it on Windows, where the newlines are not represented with `'\n'`?

Reply:
I did not. @eeckstein @compnerd Do we support swift benchmarking on Windows already?

Reply:
@compnerd I tried triggering “@swift-ci please test Windows platform” in hope that it runs also the python unit tests for benchmarks. Is that a correct assumption?

Reply:
OK, I think it does not at the moment, because of https://github.com/apple/swift/blob/master/test/benchmark/benchmark-scripts.test-sh#L2